pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1406,3 +1550,14 @@ func (nc *Controller) reconcileNodeLabels(nodeName string) error {
 	}
 	return nil
 }
+
+// createPodKey creates pod key based on pod namespace and name.
+func createPodKey(namespace, name string) string {
```

Comment:
Key doesn't have to be a string, because then you end up with this ugly splitting.
Please make a dedicated struct for it:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodelifecycle/scheduler/taint_manager.go#L59

Reply:
Done.